pygfunction/gfunction.py

```diff
@@ -1714,7 +1714,7 @@ def thermal_response_factors(self, time, alpha, kind='linear'):
                         j1 = j0 + nq_j
                     h_ij[i0:i1, j0:j1, 1:] = h[:, j0-i1:j1-i1, :]
                     if j > i:
-                        H_ratio = segment_lengths[:, i0:i1] / \
+                        H_ratio = segment_lengths[:, i0:i0+(j1-j0)] / \
```

Comment:
Both of these have to do with `method: detailed`:

- b61c2879de6be2411a523ec70c9689a9050b57e5 made it possible to define different heights by making the reciprocal calculation occur via a vector of segment height ratios
- 0a4ac88ad4ca129648ccd0e822d74e1a7d4f283f (above) makes the solution run when the number of segments on each borehole is not the same

When the number of segments of `borehole1` and `borehole2` are not the same, the segment slice lengths are unequal. I made the program run by ensuring the slice lengths are the same, but some of the ratios then become 1. I'm not sure if this solution is accurate.